src/util.js

```diff
@@ -81,44 +81,69 @@ export function phrase (me, id, ...args) {
 	return id + " " + args.join(" ");
 }
 
-/**
- * Test whether a URL matches a set of criteria.
- * @param {string | URL} url
- * @param {{protocol, host, urls} | {protocol, host, urls}[]} criteria
- * @returns {boolean}
- */
-export function testURL (url, criteria) {
-	if (typeof url === "string") {
-		url = new URL(url);
+export const URLPattern = globalThis.URLPattern ?? import("../lib/urlpattern-polyfill/index.js");
+
+export function testURLs (source, urls) {
+	if (!urls) {
+		return false;
 	}
 
-	if (Array.isArray(criteria)) {
-		return criteria.some(pattern => testURL(url, pattern));
+	for (let i=0; i<urls.length; i++) {
+		let url = urls[i];
+
+		if (!(url instanceof URLPattern)) {
+			url = urls[i] = new URLPattern(url);
+		}
```

Comment:
Will it be a performance hit if, to avoid extra checks, we build a new URLPattern object for every URL? I checked, it it doesn't throw if we instantiate a new `URLPattern` object with the existing `URLPattern` object. Or is it some anti-pattern?

```js
let url = urls[i] = new URLPattern(urls[i]);
```

Reply:
That would create a new object that is a clone of the one passed, which is a *much* slower operation than an `instanceof` check. You can verify this via:

```js
let p = new URLPattern("https://mavo.io/:part/*");
let p2 = new URLPattern(p);
p === p2
```

Reply:
I see. Thanks.